src/stepFunctions/commands/visualizeStateMachine.ts

```diff
@@ -29,257 +29,339 @@ export interface messageObject {
-        <div class="graph-buttons-container">
-            <button id="zoomin">
-                <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
-                    <line x1="8" y1="1" x2="8" y2="15"></line>
-                    <line x1="15" y1="8" x2="1" y2="8"></line>
-                </svg>
-            </button>
-            <button id="zoomout">
-                <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
-                    <line x1="15" y1="8" x2="1" y2="8"></line>
+            <div class="status-info">
+                <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
+                    <circle cx="50" cy="50" r="42" stroke-width="4" />
                 </svg>
-            </button>
-            <button id="center">
-                <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
-                    <circle cx="8" cy="8" r="7" stroke-width="2" />
-                    <circle cx="8" cy="8" r="1" stroke-width="2" />
-                </svg>
-            </button>
-        </div>
+                <div class="status-messages">
+                    <span class="previewing-asl-message">${statusTexts.inSync}</span>
+                    <span class="rendering-asl-message">${statusTexts.syncing}</span>
+                    <span class="error-asl-message">${statusTexts.notInSync}</span>
+                </div>
+            </div>
+            <div class="graph-buttons-container">
+                <button id="zoomin">
+                    <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+                        <line x1="8" y1="1" x2="8" y2="15"></line>
+                        <line x1="15" y1="8" x2="1" y2="8"></line>
+                    </svg>
+                </button>
+                <button id="zoomout">
+                    <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+                        <line x1="15" y1="8" x2="1" y2="8"></line>
+                    </svg>
+                </button>
+                <button id="center">
+                    <svg focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+                        <circle cx="8" cy="8" r="7" stroke-width="2" />
+                        <circle cx="8" cy="8" r="1" stroke-width="2" />
+                    </svg>
+                </button>
+            </div>
+
+            <script src='${webviewBodyScript}'></script>
+        </body>
+    </html>`
+    }
+
+    private trackedDocumentDoesExist(trackedDocumentURI: vscode.Uri): boolean {
+        const document = vscode.workspace.textDocuments.find(doc => doc.fileName === trackedDocumentURI.fsPath)
 
-        <script src='${webviewBodyScript}'></script>
-    </body>
-</html>`
+        return document !== undefined
+    }
+
+    private static async isDocumentValid(textDocument?: vscode.TextDocument): Promise<boolean> {
```

Comment:
This method seems to be performing general utility validation. Any reason it isn't a freestanding method outside of the class? Seems like it could be tested in isolation as well that way.

Reply:
Is there a different file that makes more sense to place this function? Or is it okay freestanding in the AslVisualization file?

Reply:
Freestanding within the same file until there is a need for it to be shared is okay

Reply:
Alan has moved this function as part of a different pull request.

Reply:
I currently am having issues with the checks passing, is there a way to run the tests locally and view the logs?